lib/core/src/Cardano/DB/Sqlite.hs

```diff
@@ -345,19 +360,38 @@ instance MatchMigrationError SqliteException where
 newtype ManualMigration = ManualMigration
     { executeManualMigration :: Sqlite.Connection -> IO () }
 
-createSqliteBackend
+createSqlitePool
     :: Tracer IO DBLog
     -> Maybe FilePath
     -> ManualMigration
     -> LogFunc
-    -> IO (SqlBackend, Sqlite.Connection)
-createSqliteBackend trace fp migration logFunc = do
+    -> IO (Pool (SqlBackend, Sqlite.Connection))
+createSqlitePool tr fp migration logFunc = do
     let connStr = sqliteConnStr fp
-    traceWith trace $ MsgConnStr connStr
-    conn <- Sqlite.open connStr
-    executeManualMigration migration conn
-    backend <- wrapConnectionInfo (mkSqliteConnectionInfo connStr) conn logFunc
-    pure (backend, conn)
+    traceWith tr $ MsgConnStr connStr
+
+    let createConnection = do
+            let info = mkSqliteConnectionInfo connStr
+            conn <- Sqlite.open connStr
+            executeManualMigration migration conn
+            backend <- wrapConnectionInfo info conn logFunc
+            pure (backend, conn)
+
+    let destroyConnection = \(backend, _) -> do
+            destroySqliteBackend tr backend fp
+
+    createPool
+        createConnection
+        destroyConnection
+        numberOfStripes
+        timeToLive
```

Comment:
If the pool thread is cancelled (i.e. wallet is deleted or server needs to exit), those lingering pool connections are going to get cleaned up promptly, right?

Reply:
Yes. I swapped the `destroyDBLayer` function with:

```hs
destroyDBLayer :: Tracer IO DBLog -> SqliteContext -> IO ()
destroyDBLayer tr SqliteContext{connectionPool,dbFile} = do
    traceWith tr (MsgDestroyConnectionPool dbFile)
    destroyAllResources connectionPool
```

So this is done as part of the `withDBLayer` bracket. The same quirks as before with regards to closing a single SQLite connection still applies but `destroyConnection` now does exactly what `destroyDBLayer` was doing, that is, retry for some cherry-picked exceptions and handle already closed connections as well.